server/src/main/java/org/candlepin/dto/manifest/v1/ProductDTO.java

```diff
@@ -846,6 +847,92 @@ private boolean isNullOrIncomplete(BrandingDTO branding) {
             branding.getType() == null || branding.getType().isEmpty();
     }
 
+    /**
+     * Retrieves a view of the provided products for the product represented by this DTO.
+     * If the provided products have not yet been defined, this method returns null.
+     *
+     * Note that the collection returned by this method is a view of the collection backing this
+     * set of provided products. Elements cannot be added to the collection, but elements may be removed.
+     * Changes made to the collection will be reflected by this product DTO instance.
+     *
+     * @return
+     *  The provided products associated with this key, or null if they have not yet been defined
+     */
+    @JsonIgnore
```

Comment:
I know you've added the ignore annotations here, but the truth is that since the manifest API should not change (provided products should still stay in the manifest PoolDTO), there is no need to add this field on this DTO at all.
I know you've added it because the ProductInfo interface dictates it, but this is another case of asking: why does this DTO need that interface??? @Ceiu since I see you added that, what was the reason? Our manifest API is clearly different than our model, our REST API, and our adapters API, so do we need that interface here?

Reply:
Something that seems to be getting lost or misunderstood is what role the various models play in conjunction with each other, and a bit of Candlepin's development history.

First, the model we care about primarily is our data model (the entities). That drives how we store data and how we work with data at the logic level. The DTOs and various DTO-like things (adapter \*Info interfaces) are views into that model or data. We try to keep them as a subset of the data layer, but that isn't always possible due to preexisting conditions, usages, or just some legacy behavior we cannot remove or update for some reason. In any event, the data model is what drives most of the design of everything around it, not the views.

In the case of the adapter interfaces, they are a stopgap implementation to get adapter consumers (IT) to stop using our internal model objects and start using their own, for both maintenance and performance reasons. However, as soon as the adapter's model changed, we had to address the fallout of how the adapters were designed and intended to be used vs how they ended up actually being used; coupled with the fact we now had code that was reliant on multiple versions of the model (data, manifest view and adapter view). So, we were left with a few options, one of which was rewrite every bit of code that surrounded the adapters (i.e. most of the manifest processing, chunks of refresh, etc.), which, as I'm sure you can imagine, was deemed unviable given the sheer volume of such work. After several discussions we landed on the interface + existing DTO reuse we have today. @kahowell might remember and have more to add here if he cares enough.

Now, should the adapters have their own DTO implementations that we use? Maybe. Given how we seem to *want* to use the adapters, probably not, though. However, both answers to this question come with plenty of baggage and need to be properly understood before a decision is made. Perhaps the DTO layers should all be interfaces and we can implement them in whatever means we choose?

So then... getting back to the original question of why do we need the JsonIgnore annotation. Short answer is: we don't. Having extra fields is a waste of data and processing time, but it's not strictly required that we avoid it if the change is a net addition that doesn't violate any security concerns. In this specific case, I'd be curious to understand why we don't want the product model

We have other potential concerns here, such as the one @nikosmoum is seemingly alluding to here and one I brought up in the design discussion yesterday in that we should be avoiding having a view dictate the model implementation. That kind of applies here, but perhaps should be better framed as we should be working to separate the shared implementation of the two interfaces if the overlap is no longer maintainable.

Reply:
I can go more in-depth on some of this stuff if necessary, but it's getting hard to keep up with given how messy the history and design behind all of this is; and the research involved just in writing this part took a good 90 minutes of my morning.

Reply:
From what I understand by looking at manifest ProductDTO which is closely tied with ProductInfo interface at multiple places involving refresh flow and manifest processing.
Are we planing to refactor this by any chance ? @Ceiu @nikosmoum 
Removing JsonIgnore as it is not strictly needed.

Reply:
@Ceiu Thanks for the detailed background. I suppose removing the jsonignore annotation is fine here, since net additions aren't a direct problem. Maybe it would be good to document it in the comment for future reference. Adding new fields to our APIs just because we can, but without actually using them is not something I'm a fan of (We should be more intentional with our API changes, but this is I guess a longer discussion that relates our overall API/versioning strategy).

> but perhaps should be better framed as we should be working to separate the shared implementation of the two interfaces if the overlap is no longer maintainable.

Yes, ideally, just like a view shouldn't dictate our model, a view (adapters API) should not dictate another view (manifest API) in my opinion.

Reply:
Something that's being glossed over in all of this is that manifests use two models simultaneously. If we spend the time to separate that out, or otherwise standardize the manifest DTO layer such that it lines up with the adapter layer (which is kind of what happens when ProductDTO inherits ProductInfo), then this mostly goes away on its own.

There are a couple ways of doing that:
- Write a translator and copy the data between the models (again)
- Decouple the manifest code from the refresh path which would drop the need for the adapter models in manifest.
- Decouple refresh from the adapter models

And so on and so forth. Lots of options to do it, but none of them are easy or efficient; and despite many pleas on my behalf, we haven't found/had time to dedicate to giving manifest the rewrite/refactor it deserves.